File: Data Structures & Algorithms/reverse-integer/submission-0.java
class Solution {
    public int reverse(int x) {
        int rev = 0;
        while(x!=0){
            int num = x%10;
            x=x/10;
            int temp = rev*10+num;
            if((temp-num)/10!=rev){
                return 0;
            }
            rev = temp;
        }
        return rev;
    }
}

File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        prev = None
        curr = head
        
        while curr:
            # Save the next node
            next_temp = curr.next
            # Reverse the link
            curr.next = prev
            # Move pointers one step forward
            prev = curr
            curr = next_temp
        
        return prev
